tuxemon/event/actions/random_battle.py

```diff
@@ -82,17 +88,17 @@ def start(self) -> None:
             current_monster.experience_modifier = level
             npc.add_monster(current_monster, len(npc.monsters))
 
-        # Don't start a battle if we don't even have monsters in our party yet.
-        if not check_battle_legal(player):
-            logger.warning("battle is not legal, won't start")
-            return
+        player = self.session.player
 
         # Lookup the environment
         env_slug = player.game_variables.get("environment", "grass")
         env = db.lookup(env_slug, table="environment")
 
+        if not check_battle_legal(player) or not check_battle_legal(npc):
```

Comment:
lets avoid removing log lines unless they are no longer useful.  in this case, i thin kits still a good idea to include 
```
logger.warning("battle is not legal, won't start")
```
when you are debugging why a combat isn't working

Reply:
ah ok, I thought the logger inside check_battle_legal was enough.